### DISCUSSION TITLE
ScriptReference/Texture2D

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#390] [Notes]
# ScriptReference/Texture2D

https://docs.unity3d.com/ScriptReference/Texture2D.html

**Comment by cmcpasserby** [comment]
Some of the old methods for loading a image from byte array as well as encoding to PNG or EXR have been removed from this calls and re added as extension methods in [`ImageConversion`](https://docs.unity3d.com/ScriptReference/ImageConversion.html)

**Comment by YoavTC** [comment]
`Texture2D.whiteTexture` (and the others as well) are not protected at all, and Unity uses them to draw the Editor UI. 
Modifying `Texture2D.whiteTexture` at runtime modifies the editor as well, and it stays even after exiting play mode.

Running this should reset everything:

```
var texture2D = Texture2D.whiteTexture;

var pixels = texture2D.GetPixels();
for (int i = 0; i < pixels.Length; i++)
{
        pixels[i] = Color.white;
}
            
texture2D.SetPixels(pixels);
texture2D.Apply();
```

Not sure if the dark mode uses the `Texture2D.blackTexture` though.
